q2_types_genomics/eggnog/_validator.py

```diff
@@ -0,0 +1,56 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+from ..plugin_setup import plugin
+from qiime2.core.exceptions import ValidationError
+
+import pandas as pd
+from q2_types.feature_data import FeatureData
+from q2_types_genomics.eggnog import (NOG, OG, KEGG, )
+
+nog_fields = set(['query_name', 'seed_eggNOG_ortholog',
+                  'seed_ortholog_evalue', 'seed_ortholog_score',
+                  'eggNOG OGs', 'narr_og_name', 'narr_og_cat',
+                  'narr_og_desc', 'best_og_name', 'best_og_cat',
+                  'best_og_desc', 'Preferred_name', 'GOs', 'EC',
+                  'KEGG_ko', 'KEGG_Pathway', 'KEGG_Module', 'KEGG_Reaction',
+                  'KEGG_rclass', 'BRITE', 'KEGG_TC', 'CAZy',
+                  'BiGG_Reaction', 'PFAMs',
+                  ])
+
+og_fields = set(['eggNOG', 'OGs', 'narr_og_name', 'narr_og_cat',
+                 'nar_og_desc',
+                 ])
+
+kegg_fields = set(['KEGG_ko', 'KEGG_Pathway', 'KEGG_Module', 'KEGG_Reaction',
+                   'KEGG_rclass', 'BRITE', 'KEGG_TC',
+                   ])
+
+
+@plugin.register_validator(FeatureData[NOG])
+def validate_nog(data: pd.DataFrame, level):
+    _check_fields(data=data, level=level, reference_fields=nog_fields)
+
+
+@plugin.register_validator(FeatureData[OG])
+def validate_og(data: pd.DataFrame, level):
+    _check_fields(data=data, level=level, reference_fields=og_fields)
+
+
+@plugin.register_validator(FeatureData[KEGG])
+def validate_kegg(data: pd.DataFrame, level):
+    _check_fields(data=data, level=level, reference_fields=kegg_fields)
+
+
+def _check_fields(data: pd.DataFrame, level, reference_fields):
```

Comment:
Does this function need the `level` arg? Unless you anticipate some usage in the near future, please remove.